src/Pagination/index.scss

```diff
@@ -1,18 +1,30 @@
 .nds-pagination {
+  --page-min-size: 1.5em;
+  color: var(--theme-primary);
   display: flex;
-  align-items: center;
   justify-content: center;
-  color: var(--theme-primary);
+  align-items: center;
 
-  ul {
-    list-style-type: none;
-  }
   ul,
   li {
     text-indent: 0;
     margin: 0;
     padding: 0;
   }
+
+  ul {
+    list-style-type: none;
+    display: grid;
+    grid-auto-flow: column;
+    justify-items: center;
+    align-items: center;
+    gap: var(--space-xxs);
+
+    // Every column should be the same width as the widest item to prevent
+    // UX shifting when paging between single, double, tripple, (or more)
+    grid-auto-columns: 1fr;
+    width: max-content;
```

Comment:
For every column created for an item, make it `1fr` of total columns, which divides the available space equally. Each column is set to max-width, so the widest item will participate in the meaning of `1fr`. This is a trait unique to CSS grid